pkg/net/libp2p/key_test.go

```diff
@@ -23,25 +25,37 @@ func TestOperatorPrivateKeyToNetworkKeyPair(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(networkPublicKey.Curve.Params(), DefaultCurve.Params()) {
-		t.Errorf("network public key uses the wrong curve")
+	secp256k1PublicKey, ok := networkPublicKey.(*libp2pcrypto.Secp256k1PublicKey)
+	if !ok {
+		t.Fatal("wrong type of public key")
 	}
 
-	if networkPublicKey.X.Cmp(operatorPublicKey.X) != 0 {
+	btcecPublicKey := (*btcec.PublicKey)(secp256k1PublicKey)
+
+	if btcecPublicKey.X().Cmp(operatorPublicKey.X) != 0 {
 		t.Errorf("network public key has a wrong X coordinate")
 	}
-	if networkPublicKey.Y.Cmp(operatorPublicKey.Y) != 0 {
+	if btcecPublicKey.Y().Cmp(operatorPublicKey.Y) != 0 {
 		t.Errorf("network public key has a wrong Y coordinate")
 	}
 
+	secp256k1PrivateKey, ok := networkPrivateKey.(*libp2pcrypto.Secp256k1PrivateKey)
+	if !ok {
+		t.Fatal("wrong type of private key")
+	}
```

Comment:
```suggestion
	libp2pPrivateKey, ok := networkPrivateKey.(*libp2pcrypto.Secp256k1PrivateKey)
	if !ok {
		t.Fatal("network private key is not an instance of the libp2p secp256k1 private key")
	}
```

Reply:
Type assertions were removed from this test in https://github.com/keep-network/keep-core/commit/027febb4837f04ec11bc4e208c0b1e28a8da3642.